src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const statesHistory = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        currentState = { ...currentState, ...extraData };
+        break;
+
+      case 'removeProperties':
+        currentState = { ...currentState };
+
+        for (const key of keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+    }
```

Comment:
[CODE STYLE]: The `switch` statement is missing a `default` case. It's a good practice to include a `default` case for handling unexpected `type` values. This can be useful for error handling and to ensure that the function behaves correctly even if it receives an action with an unknown type.